visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -1,39 +1,46 @@
-import { FileValidator, CCValidationResult } from "./util/fileValidator"
+import { validate, CCValidationResult } from "./util/fileValidator"
 import { AttributeTypes, CCFile, NameDataPair, BlacklistType, BlacklistItem } from "./codeCharta.model"
 import _ from "lodash"
 import { NodeDecorator } from "./util/nodeDecorator"
+import { OldAttributeTypes, ExportBlacklistType, ExportCCFile } from "./codeCharta.api.model"
 import { StoreService } from "./state/store.service"
 import { addFile, setSingle } from "./state/store/files/files.actions"
 import { getCCFiles } from "./model/files/files.helper"
+import { DialogService } from "./ui/dialog/dialog.service"
 
 export class CodeChartaService {
 	public static ROOT_NAME = "root"
 	public static ROOT_PATH = "/" + CodeChartaService.ROOT_NAME
 	public static readonly CC_FILE_EXTENSION = ".cc.json"
 
-	constructor(private storeService: StoreService) {}
+	constructor(private storeService: StoreService, private dialogService: DialogService) {}
 
 	public loadFiles(nameDataPairs: NameDataPair[]): Promise<void> {
 		return new Promise((resolve, reject) => {
 			nameDataPairs.forEach((nameDataPair: NameDataPair) => {
-				const validationResult: CCValidationResult = FileValidator.validate(nameDataPair.content)
-				if (validationResult.error.length === 0) {
-					const ccFile = this.getCCFile(nameDataPair.fileName, nameDataPair.content)
-					NodeDecorator.preDecorateFile(ccFile)
-					this.storeService.dispatch(addFile(ccFile))
-					if (validationResult.warning.length !== 0) {
-						reject(validationResult)
+				try {
+					validate(nameDataPair.content)
+				} catch (e) {
+					if (e.warning.length > 0) {
+						this.printWarnings(e)
```

Comment:
Using `try / catch` here seems the wrong approach.

There are three ways I see how to handle this: either return `warnings` only from the validation function and `throw` inside the validation function in case of actual errors. That way the try / catch is not necessary.
An alternative would be to pass through the `printWarnings` function to the validate function, so that it would be triggered from inside the validate function.
The third one is to split the validation function up into pure `errors` and then check for warnings in a separate step afterwards.

Reply:
Your first option is implies to return a validation-result containing the warnings. In another comment you said, that returning something in a validate-function is unexpected, since it should just validate and throw errors.

I think we should get rid of the warnings in the validation and handle them separately after we know that our file is valid.